buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlTestPlugin.java

```diff
@@ -0,0 +1,126 @@
+                task -> { task.dependsOn(project.getTasks().getByName("copyRestApiSpecsTask")); }
+            );
+        yamlTestTask.setGroup(JavaBasePlugin.VERIFICATION_GROUP);
+        yamlTestTask.setDescription("Runs the YAML based REST tests against an external cluster");
+
+        // setup task dependency
+        project.getDependencies().add(SOURCE_SET_NAME + "Compile", project.project(":test:framework"));
+
+        // setup the runner
+        RestTestRunnerTask runner = (RestTestRunnerTask) project.getTasks().getByName(yamlTestTask.getName() + "Runner");
+        runner.setTestClassesDirs(yamlTestSourceSet.getOutput().getClassesDirs());
+        runner.setClasspath(yamlTestSourceSet.getRuntimeClasspath());
+
+        // if this a module or plugin, it may have an associated zip file with it's contents, add that to the test cluster
+        boolean isModule = project.getPath().startsWith(":modules:");
+        Zip bundle = (Zip) project.getTasks().findByName("bundlePlugin");
+        if (bundle != null) {
+            yamlTestTask.dependsOn(bundle);
+            if (isModule) {
+                runner.getClusters().forEach(c -> c.module(bundle.getArchiveFile()));
+            } else {
+                runner.getClusters().forEach(c -> c.plugin(project.getObjects().fileProperty().value(bundle.getArchiveFile())));
+            }
+        }
+
+        // es-plugins may declare dependencies on additional modules, add those to the test cluster too.
+        project.afterEvaluate(p -> {
+            PluginPropertiesExtension pluginPropertiesExtension = project.getExtensions().findByType(PluginPropertiesExtension.class);
+            if (pluginPropertiesExtension != null) { // not all projects are defined as plugins
+                pluginPropertiesExtension.getExtendedPlugins().forEach(pluginName -> {
+                    Project extensionProject = project.getProject().findProject(":modules:" + pluginName);
+                    if (extensionProject != null) { // extension plugin may be defined, but not required to be a module
+                        Zip extensionBundle = (Zip) extensionProject.getTasks().getByName("bundlePlugin");
+                        yamlTestTask.dependsOn(extensionBundle);
+                        runner.getClusters().forEach(c -> c.module(extensionBundle.getArchiveFile()));
+                    }
+                });
+            }
+        });
+
+        // setup IDE
+        GradleUtils.setupIdeForTestSourceSet(project, yamlTestSourceSet);
+
+        // run test tasks first if they exist since they are presumably faster and less resources intensive
+        Task testTask = project.getTasks().findByName("test");
+        if (testTask != null) {
+            yamlTestTask.mustRunAfter(testTask);
+        }
+
+        // validation of the rest specification is wired to precommit, so ensure that runs first
+        yamlTestTask.mustRunAfter(project.getTasks().getByName("precommit"));
```

Comment:
What's creating the `precommit` task in this scenario? Is it some other plugin we're applying?

Reply:
Note https://github.com/elastic/elasticsearch/pull/56926 will make this easier, you shouldn't have to worry about it. Currently that we already have:

```
project.getTasks().named(JavaPlugin.TEST_TASK_NAME).configure(t -> t.mustRunAfter(precommit));
```

Reply:
I removed this line since #56926 addressed all Test tasks